src/lib.rs

```diff
@@ -218,6 +218,35 @@ pub struct BytesError {
     found: usize,
 }
 
+/// A general error that can occur when handling [`Uuid`]s.
+///
+/// Although specialized error types exist in the crate,
+/// sometimes where particular error type occurred is hidden
+/// until errors need to be handled. This allows to enumerate 
+/// the errors.
+///
+/// [`Uuid`]: struct.Uuid.html
+// TODO: improve the doc
+// BODY: This detail should be fine for initial merge
+#[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+pub enum Error {
```

Comment:
I think we'd be better off with an opaque struct here than an `enum`. Something like:

```rust
pub struct Error(Inner);

enum Inner {
    Bytes(..),
    Parse(..),
}

impl Error {
    pub fn is_bytes(&self) -> bool {
        ..
    }

    pub fn is_parse(&self) -> bool {
        ..
    }
}
```

And making the `BytesError` and `ParseError` types private in our next breaking release in favour of this single error type. I don't think we need a complex error management story for `uuid`.

Reply:
That does seem like a good idea...

Reply:
A note on the making the `BytesError` and `parser::ParseError` private: I designed the parse error type in a way so crates like `blog-uuid` and `uuid-b64` can it and have a seamless design. Should I abandon that or should I readjust API accordingly?

Reply:
@KodrAus

Reply:
i would prefer readjusting the API first if possible

Reply:
@kinggoesgaming Ah so that other libraries can try parse a uuid and use the returned error type to maybe build up their own more semantic error to show a user?

Reply:
@KodrAus that is a good way to put it. :smile:

Reply:
I will open this PR for review... if we need to alter the API as the way you suggested @KodrAus, it can be done in `0.8.0` as it will affect the error APIs anyways